### PR TITLE
fix(curriculum): separate unreachable code and scope for pyramid project

### LINKED ISSUE #54809: Teach unreachable code and scope in separate lessons for pyramid project
### Describe the Issue

on starting step 51 of the JS Beta project to build a pyramid generator, I noticed that we're being asked to add a variable declaration and that the editor has an empty line within the padRow function but -after- the return statement. As these blank lines are meant to be suggestive to learners (nudging them as to where to type), this empty line after the return seemed to be badly placed as it shouldn't suggest that placing code after a return statement is acceptable (or that it could work with the console log in the local scope for eg which also would not work in this case)
Suggest to place the empty line within the function above the return statement instead.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-introductory-javascript-by-building-a-pyramid-generator/step-51

### Your code

N/A

### Expected behavior

empty line within padRow function should be placed above the return statement (not below it)

### Screenshots

_No response_

### System

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Supravisor**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #54809

image

image

image

Step 51 success message
image

image

image

Step52 success message
image

image

image

image

Step 53 success message
image

image

image

image

image

image

Step 54 success message
image

>UPDATED Step 53
image

>UPDATED Step 54
image

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c77d50636722e5b6be17.md [hunk 1]
```diff
@@ -7,9 +7,14 @@ dashedName: step-53
 
 # --description--
 
-If you try to add a `console.log(test)` call below your `padRow` function, you would see an error. This is because `test` is defined in the local scope, meaning you cannot access it in the global scope (outside of the `padRow` function).
+Add `console.log("This works!");` below the `return` statement.
```
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c77d50636722e5b6be17.md:10]:
```suggestion
Below the `return` statement, log the string `"This works!"` to the console.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c77d50636722e5b6be17.md [hunk 2]
```diff
@@ -7,9 +7,14 @@ dashedName: step-53
...
 
-Returning a value from a function brings that value into the scope where the function was called. To bring your `"Testing"` value from the `padRow` function into the global scope, update your `return` statement to return only the `test` variable.
+There is a bit of a problem, here. Your `console.log` is currently after your `return` statement.
+The string `"This works!"` does not display in the console, and the newly placed code appears washed out.
+Copy and paste the `console.log` above the `return` statement.
+Now the string `"This works!"` appears in the console.
+
+An important thing to know about the `return` keyword is that it does not just define a value to be returned from your function, it also stops the execution of your function code. This means that any code after a `return` statement will not run.
```
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c77d50636722e5b6be17.md:17]:
```suggestion
After doing that, you will see that the string `"This works!"` does not display in the console, and the `console.log("This works!")` line is greyed out.

Copy and paste the line of code above the `return` statement. Now, the string `"This works!"` should appear in the console.

An important thing to know about the `return` keyword is that it does not just define a value to be returned from your function, it also stops the execution of your code inside a function or a block statement. This means any code after a `return` statement will not run.
```

**Review by Ksound22 — APPROVED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c6541c82551f95e765ab.md
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c6541c82551f95e765ab.md:1]: Can you please edit the text in step 51 (now 53)

> If you try to access it outside of the function, you will either get `undefined` or an error.

To

> If you try to access it outside of the function, you get a reference error.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c71600966a2191d3a64a.md [hunk 3]
```diff
@@ -7,17 +7,14 @@ dashedName: step-52
 
 # --description--
 
-There is a bit of a problem, here. Your `test` declaration is currently after your `return` statement. An important thing to know about the `return` keyword is that it does not just define a value to be returned from your function, it also stops the execution of your function code. This means that any code after a `return` statement will not run.
-
-Move your `test` initialization to the line above your `return` statement.
+Returning a value from a function brings that value into the scope where the function was called. To bring your `"Testing"` value from the `padRow` function into the global scope, update your return statement to return only the `test` variable.
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c71600966a2191d3a64a.md:10]:
````md
Values returned out of functions are used by calling the function. You can use the function call directly as the value it returns, or capture the returned value in a variable. This way, you can use the value assigned to a locally scoped variable, outside the function it was created in.

```js
function getName() {
  const name = "Camper cat";
  return name;
}

console.log(getName()); // "Camper cat"

const capturedReturnValue = getName();
console.log(capturedReturnValue); // "Camper cat"

console.log(name); // reference error
```

To use your `"Testing"` value returned from the `padRow` function, update your `return` statement to return only the `test` variable.

````

---

This is just a suggestion, but I think what we have now can be misinterpreted in multiple ways. I suggest we simplify this and give a small code example.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c77d50636722e5b6be17.md [hunk 4]
```diff
@@ -7,9 +7,13 @@ dashedName: step-53
 
 # --description--
 
-If you try to add a `console.log(test)` call below your `padRow` function, you would see an error. This is because `test` is defined in the local scope, meaning you cannot access it in the global scope (outside of the `padRow` function).
+Below the `return` statement, log the string `"This works!"` to the console.
 
-Returning a value from a function brings that value into the scope where the function was called. To bring your `"Testing"` value from the `padRow` function into the global scope, update your `return` statement to return only the `test` variable.
+After doing that, you will see that the string `"This works!"` does not display in the console, and the `console.log("This works!")` line is greyed out.
+
+Copy and paste the line of code above the `return` statement. Now, the string `"This works!"` should appear in the console.
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c77d50636722e5b6be17.md:14]:
```suggestion
Copy the console log and paste it above the `return` statement. Now, the string `"This works!"` should appear in the console.
```

**Review by lasjorg — CHANGES_REQUESTED** [review]
Two new challenges were added, so step 51 is now 53.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c71600966a2191d3a64a.md [hunk 5]
```diff
@@ -7,17 +7,31 @@ dashedName: step-54
...
+
+console.log(getName()); // "Camper cat"
+
+const capturedReturnValue = getName();
+console.log(capturedReturnValue); // "Camper cat"
+
+console.log(name); // reference error
+```
+
+To use your `"Testing"` value returned from the `padRow` function, update your `return` statement to return only the `test` variable.
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/6610c71600966a2191d3a64a.md:26]:
```suggestion
To use your `"Testing"` value, return it out of the `padRow` function by updating your `return` statement to return only the `test` variable.
```

Sorry, I missed this when I updated the text.

**Review by lasjorg — CHANGES_REQUESTED** [review]

**Review by lasjorg — APPROVED** [review]
This looks good.

---

I was thinking that we could try to avoid the assert message for the two console logs in Step 55 if the camper typos the log string (just to avoid confusion) but I don't think we need to do that.

**Pull request merged**